webui/react/src/components/ExperimentActionDropdown.tsx

```diff
@@ -98,6 +104,17 @@ const ExperimentActionDropdown: React.FC<Props> = ({
           openCommand(tensorboard);
           break;
         }
+        case Action.Pin: {
+          const pinList = [ ...settings.pinned ];
+          pinList.push(id);
+          updateSettings({ pinned: [ ...Array.from(new Set(pinList)) ] });
```

Comment:
perf: I didn't benchmark this one, and my gut says it's less important since you're not filtering the array like you are in unpin, but the following will still probably be faster:
```
const pinList = new Set(settings.pinned);
pinList.add(id);
updateSettings({ pinned: [ ...pinList ] });